components/Common/DarkModeToggle/index.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect } from 'react';
+import ModeNightIcon from '@mui/icons-material/ModeNight';
+import BrightnessMediumIcon from '@mui/icons-material/BrightnessMedium';
+import styles from './index.module.scss';
+import { useTheme } from 'next-themes';
+
+const THEME = {
+  DARK: 'dark',
+  LIGHT: 'light',
+} as const;
+
+function updateBodyClass(theme: string) {
+  if (typeof document === 'undefined' || !theme) return;
+  const checkTheme = theme === THEME.DARK ? THEME.LIGHT : THEME.DARK;
+  // Remove the previous theme classname before adding
+  if (document.body.classList.contains(checkTheme)) {
+    document.body.classList.remove(checkTheme);
+  }
+  document.body.classList.add(theme);
+}
+
+const DarkModeToggle = () => {
+  const { theme, setTheme } = useTheme();
+
+  const isDark = theme === 'dark';
+
+  useEffect(() => {
+    if (theme) updateBodyClass(theme);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  function toggleTheme(isKeyPress?: boolean) {
```

Comment:
This should be an arrow function.

Reply:
what's wrong with a normal `function` 🙃

Reply:
It doesn't make sense to write a named function within a react component when you're using a functional component with an arrow function approach. It's an antipattern.